### PR TITLE
db/confirmation-popup-refactor

### PR DESCRIPTION
**Pull request by karolkolodziej**
Refactored confirmation popup to use the `BaseForm`.

### CODE REVIEW

File: css/dashboards/gui.css [hunk 1]
```diff
@@ -749,28 +746,59 @@
...
     width: 200px;
     left: calc(50% - 150px);
+    display: none;
+    background-color: var(--highcharts-dashboards-main-background-color);
+    color: var(--highcharts-neutral-color-60);
+    font-size: 0.876em;
+    max-height: 90%;
+    top: 5%;
+    overflow-x: hidden;
+    overflow-y: hidden;
```
Comment (sebastianbochan) [inline css/dashboards/gui.css:762]: Replace
```
overflow-x: hidden;
overflow-y: hidden;
```
with
```
overflow:hidden
```

File: css/dashboards/gui.css [hunk 2]
```diff
@@ -749,28 +746,59 @@
...
+    min-width: 300px;
+    max-width: 600px;
```
Comment (sebastianbochan) [inline css/dashboards/gui.css:764]: In case of `width: 200px;` the `max-width` will not be used anymore.

File: css/dashboards/gui.css [hunk 3]
```diff
@@ -749,28 +746,59 @@
...
     left: calc(50% - 150px);
+    display: none;
+    background-color: var(--highcharts-dashboards-main-background-color);
+    color: var(--highcharts-neutral-color-60);
+    font-size: 0.876em;
+    max-height: 90%;
+    top: 5%;
+    overflow-x: hidden;
+    overflow-y: hidden;
+    min-width: 300px;
```
Comment (sebastianbochan) [inline css/dashboards/gui.css:763]: The `min-width` is bigger than `width`. We should use the only one.

File: css/dashboards/gui.css [hunk 4]
```diff
@@ -749,28 +746,59 @@
...
 
-.highcharts-popup .highcharts-dashboards-edit-popup-content {
+.highcharts-dashboards-edit-popup-close {
+    background-repeat: no-repeat;
+    background-position: 50% 50%;
+    width: 20px;
+    height: 20px;
+    cursor: pointer;
+    position: absolute;
+    padding: 10px;
```
Comment (sebastianbochan) [inline css/dashboards/gui.css:779]: In case of background-position 50/50%, you can declare only width/height. The padding is unecessary.

**Review by sebastianbochan — CHANGES_REQUESTED** [review]

File: css/dashboards/gui.css [hunk 5]
```diff
@@ -749,28 +746,59 @@
     opacity: 0.3;
 }
 
-.highcharts-popup {
+/* Confirmation popup */
+
+.highcharts-dashboards-edit-popup {
     min-height: 160px;
     width: 200px;
     left: calc(50% - 150px);
```
Comment (pawellysy) [inline css/dashboards/gui.css:754]: maybe instead of this left, we could go with:
```css
left: 50%
transform: translate(-50%, 0)
```

**Review by pawellysy — CHANGES_REQUESTED** [review]

File: ts/Dashboards/EditMode/EditMode.ts [hunk 6]
```diff
@@ -99,8 +99,9 @@ class EditMode {
         this.createTools();
 
         this.confirmationPopup = new ConfirmationPopup(
-            this,
             board.container,
+            EditGlobals.iconsURL,
```
[the remark this answers is not in the document]
Comment (karolkolodziej) [reply ts/Dashboards/EditMode/EditMode.ts:103]: Any suggestions?

**Review by sebastianbochan — APPROVED** [review]

File: ts/Dashboards/EditMode/EditMode.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (pawellysy) [reply ts/Dashboards/EditMode/EditMode.ts:103]: Can be referenced as a value in the Class IMO.

[the remark this answers is not in the document]
Comment (pawellysy) [reply ts/Dashboards/EditMode/EditMode.ts:103]: nvm, leave as it is

**Review by pawellysy — APPROVED** [review]

**Pull request merged**